external/storm-kafka/src/jvm/org/apache/storm/kafka/SpoutConfig.java

```diff
@@ -37,6 +37,8 @@
     public long retryInitialDelayMs = 0;
     public double retryDelayMultiplier = 1.0;
     public long retryDelayMaxMs = 60 * 1000;
+    public int retryLimit = Integer.MAX_VALUE;
```

Comment:
This is probably an unlikely edge case, but theoretically possible: If a tuple is retried `Integer.MAX_VALUE` times, it will be dropped.

To truly turn off retry limits, I would suggest setting this to `-1` and adding logic to check for that in logic that checks to see if the retry limit has been reached.